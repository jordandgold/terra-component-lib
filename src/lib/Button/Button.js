import React from "react";

import "./Button.scss";

const Button = props => {
  const { onClick, className, text } = props;

  return (
    <div>
      <button
        onClick={e => onClick(e)}
        className={`ter-button ter-button--${className}`}
      >
        {text}
      </button>
    </div>
  );
};

export default Button;
