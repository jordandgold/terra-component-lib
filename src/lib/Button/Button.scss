.ter-button {
  box-sizing: border-box;
  font-size: 16px;
  padding: 8px 16px;
  background-color: #363545;
  color: #f3f3f7;
}
