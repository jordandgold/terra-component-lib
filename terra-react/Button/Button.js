import React from "react";
import "./Button.scss";

var Button = function Button(props) {
  var _onClick = props.onClick,
      className = props.className,
      text = props.text,
      name = props.name;
  return React.createElement(React.Fragment, null, React.createElement("button", {
    onClick: function onClick(e) {
      return _onClick(e);
    },
    className: "ter-button ".concat(className),
    name: name
  }, text));
};

export default Button;