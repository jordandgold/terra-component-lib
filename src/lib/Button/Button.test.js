import React from "react";

import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  let wrapper;
  let mockOnClick;
  let mockEventObject;

  beforeEach(() => {
    mockOnClick = jest.fn();
    wrapper = shallow(<Button onClick={mockOnClick} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call onClick method when clicked", () => {
    mockEventObject = {
      target: {
        innerHTML: "hello"
      }
    };
    wrapper.find(".ter-button").simulate("click", mockEventObject);

    expect(mockOnClick).toHaveBeenCalledWith(mockEventObject);
  });
});
