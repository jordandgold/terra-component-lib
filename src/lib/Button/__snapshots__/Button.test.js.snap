// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="ter-button undefined"
        onClick={[Function]}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "ter-button undefined",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="ter-button undefined"
          onClick={[Function]}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "ter-button undefined",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
